### ISSUE TITLE
Eventual assertion failure when feeding in large files

### ISSUE THREAD

**Issue by 5225225** [remram44/cdchunking-rs#5]
This code reproduces the bug quickly on my machine.

```rust
extern crate cdchunking;

use cdchunking::{Chunker, ZPAQ};

fn main() {
    let chunker = Chunker::new(ZPAQ::new(8));
    // It fails quicker on a smaller chunk size, but fails on any chunk size

    let random = std::fs::File::open("/dev/urandom").unwrap();

    for chunk in chunker.whole_chunks(random) {}
}
```

```rust
thread 'main' panicked at 'assertion failed: self.status != EmitStatus::AtSplit', /home/jack/.cargo/registry/src/github.com-1ecc6299db9ec823/cdchunking-0.2.0/src/lib.rs:322:13
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at libstd/sys_common/backtrace.rs:59
             at libstd/panicking.rs:380
   3: std::panicking::default_hook
             at libstd/panicking.rs:396
   4: std::panicking::rust_panic_with_hook
             at libstd/panicking.rs:576
   5: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:537
   6: <cdchunking::ChunkStream<R, I>>::read
             at /home/jack/src/test/<panic macros>:3
   7: <cdchunking::WholeChunks<R, I> as core::iter::iterator::Iterator>::next
             at /home/jack/.cargo/registry/src/github.com-1ecc6299db9ec823/cdchunking-0.2.0/src/lib.rs:272
   8: testx::main
             at src/main.rs:11
   9: std::rt::lang_start::{{closure}}
             at /checkout/src/libstd/rt.rs:74
  10: std::panicking::try::do_call
             at libstd/rt.rs:59
             at libstd/panicking.rs:479
  11: __rust_maybe_catch_panic
             at libpanic_unwind/lib.rs:102
  12: std::rt::lang_start_internal
             at libstd/panicking.rs:458
             at libstd/panic.rs:358
             at libstd/rt.rs:58
  13: std::rt::lang_start
             at /checkout/src/libstd/rt.rs:74
  14: main
  15: __libc_start_main
  16: _start
```

**Comment by remram44** [comment]
Thanks for the report! I will investigate.

**Comment by remram44** [comment]
I think 95979b90 fixes it. 8c353d56 adds a crash-test from a random stream like yours (using the `rand` crate instead of `/dev/urandom`).

**Comment by 5225225** [comment]
Yep, that seems to have fixed it. Thanks!

**Comment by remram44** [comment]
Released 0.2.1!
